ddl/partition.go

```diff
@@ -2457,16 +2457,27 @@ func (w *worker) onExchangeTablePartition(d *ddlCtx, t *meta.Meta, job *model.Jo
 				return ver, errors.Trace(err)
 			}
 		}
+		var ptInfo []schemaIDAndTableInfo
+		if len(nt.Constraints) > 0 {
+			pt.ExchangePartitionInfo = &model.ExchangePartitionInfo{
+				ExchangePartitionTableID:     nt.ID,
+				ExchangePartitionPartitionID: defID,
+			}
+			ptInfo = append(ptInfo, schemaIDAndTableInfo{
+				schemaID: ptSchemaID,
+				tblInfo:  pt,
+			})
+		}
 		nt.ExchangePartitionInfo = &model.ExchangePartitionInfo{
-			ExchangePartitionID:    ptID,
-			ExchangePartitionDefID: defID,
+			ExchangePartitionTableID:     ptID,
+			ExchangePartitionPartitionID: defID,
```

Comment:
Not sure `ExchangePartitionPartitionID` is better than `ExchangePartitionPartitionID`?
```suggestion
			ExchangePartitionDefID: defID,
```